added tcpip_copy file and tests for it

itzhak opened the pull request.

itzhak committed:
added tcpip_copy file and tests for it

kolmanv commented:
Waiting for fixes :)

itzhak committed:
add files and fixed some errors and styling
 On branch tcp_copy
 Changes to be committed:
	new file:   examples/file_copy/files_to_send/file_to_send.txt
	modified:   examples/file_copy/receiver.rb
	modified:   examples/file_copy/sender.rb
	modified:   lib/file_copy/tcpip_copy.rb
	renamed:    test/file_copy/tcpip_test.rb -> test/file_copy/tcpip_receiver_test.rb
	modified:   test/file_copy/test_files/file_to_send.txt

itzhak committed:
n tcpip_receiver.rb test
 On branch tcp_copy
 Your branch is ahead of 'origin/tcp_copy' by 1 commit.
	modified:   test/file_copy/tcpip_receiver_test.rb

kolmanv commented on test/file_copy/test_files/file_to_send.txt:
```diff
@@ -0,0 +1 @@
+this text send from file
```
Remove this file. Use StringIO inline (inline means inside the spec code).

kolmanv commented on test/file_copy/tcpip_receiver_test.rb:
```diff
@@ -0,0 +1,81 @@
+require 'test/unit'
```
If you rewrite this code, make it a spec.
Put it under spec/file_copy/...

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+require 'thread'
+
+require 'log'
+
```
Remove blank line.

itzhak replied:
done

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+    # host that send datat and files
+    class Sender
+      def initialize(addr, port_num)
+        @LOG = Log
```
Why do you use @LOG? Redundant use Log instead.

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+              queue.push unmarshaled_data
+            end
+            @LOG.info "connection #{peer_addr.join(":")} is closed by other side"
+          end # Thread.start(@server.accept) do |client|
```
Parenthesis not needed.

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+            @LOG.info("connected to peer: #{peer_addr.join(":")}")
+            while size = client.recv(4).unpack("l")[0]
+              @LOG.debug3 "begin receiving data with size #{size}"
+              data = client.recv(size)
```
Parenthesis not needed. Apply to all code where possible.

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+      def send_file(file_name)
+        #check if source file is exist
+        if (File.exists?(file_name) == false)
+          @LOG.warning("source file \'#{file_name} \' Does not exist")
```
I don't think slash is needed here because the string is using double quotes.
" kuku 'kaka' kuku" will be ok. Please check.

itzhak replied:
done

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+# Author: Itzhak B.
```
Very good!

Beside few syntax fixed the code styling is good.

kolmanv commented on lib/file_copy/tcpip_copy.rb:
```diff
@@ -0,0 +1,86 @@
+      end
+
+      def send(data)
+        @LOG.debug3("data to send is (#{data}).")
```
Parenthesis everywhere :)

kolmanv commented on examples/file_copy/sender.rb:
```diff
@@ -0,0 +1,32 @@
+require_relative '../../lib/file_copy/tcpip_copy.rb'
+
+# TODO (itzhak and yaron) check why log file is not created
```
Example file should be without module names. Remove module BBFS and module FileCopy.
Add many comments to person which sees this in the first time.
Look here:
https://github.com/kolmanv/bbfs/blob/master/examples/file_indexing.rb

kolmanv commented on examples/file_copy/files_to_send/file_to_send.txt:
```diff
@@ -0,0 +1 @@
+this is send from file located at examples/file_copy/files_to_send/file_to_send.txt
```
Make it StringIO inline. remove those small files.

itzhak committed:
t final version, inserted only for Backup
 On branch tcp_copy
	modified:   examples/file_copy/receiver.rb
	modified:   examples/file_copy/sender.rb
	modified:   lib/file_copy/tcpip_copy.rb
	modified:   test/file_copy/tcpip_receiver_test.rb

itzhak committed:
Merge branch 'master' into tcp_copy

itzhak committed:
Styling modification.
It is not final version. There is styling fixing that should be added.
 On branch tcp_copy
	modified:   examples/file_copy/receiver.rb
	modified:   examples/file_copy/sender.rb
	modified:   lib/file_copy/tcpip_copy.rb

itzhak committed:
not final version
 On branch tcp_copy
	modified:   README
	modified:   bin/backup_server
	modified:   bin/content_server
	modified:   lib/content_server.rb
	modified:   lib/content_server/queue_indexer.rb
	modified:   lib/file_copy/copy.rb
	modified:   lib/file_copy/tcpip_copy.rb
	modified:   lib/file_monitoring/file_monitoring.rb
	modified:   lib/file_monitoring/monitor_path.rb

itzhak committed:
add comment for thing to fix (TODO)
 On branch tcp_copy
	modified:   lib/file_copy/tcpip_copy.rb

itzhak committed:
Merge branch 'tcp_copy' of https://github.com/kolmanv/bbfs into tcp_copy
add todo's for fixing codes
Conflicts:
	lib/file_copy/tcpip_copy.rb

	modified:   lib/file_copy/tcpip_copy.rb
